projects/packages/identity-crisis/src/class-identity-crisis.php

```diff
@@ -0,0 +1,1026 @@
+			array(
+				'sync_error_idc',
+				'safe_mode_confirmed',
+				'migrate_for_idc',
+			)
+		);
+	}
+
+	/**
+	 * Checks whether the sync_error_idc option is valid or not, and if not, will do cleanup.
+	 *
+	 * @return bool
+	 * @since 5.4.0 Do not call get_sync_error_idc_option() unless site is in IDC
+	 *
+	 * @since 4.4.0
+	 */
+	public static function validate_sync_error_idc_option() {
+		$is_valid = false;
+
+		// Is the site opted in and does the stored sync_error_idc option match what we now generate?
+		$sync_error = Jetpack_Options::get_option( 'sync_error_idc' );
+		if ( $sync_error && self::sync_idc_optin() ) {
+			$local_options = self::get_sync_error_idc_option();
+			// Ensure all values are set.
+			if ( isset( $sync_error['home'] ) && isset( $local_options['home'] ) && isset( $sync_error['siteurl'] ) && isset( $local_options['siteurl'] ) ) {
+				// If the WP.com expected home and siteurl match local home and siteurl it is not valid IDC.
+				if (
+					isset( $sync_error['wpcom_home'] ) &&
+					isset( $sync_error['wpcom_siteurl'] ) &&
+					$sync_error['wpcom_home'] === $local_options['home'] &&
+					$sync_error['wpcom_siteurl'] === $local_options['siteurl']
+				) {
+					$is_valid = false;
+					// Enable migrate_for_idc so that sync actions are accepted.
+					Jetpack_Options::update_option( 'migrate_for_idc', true );
+				} elseif ( $sync_error['home'] === $local_options['home'] && $sync_error['siteurl'] === $local_options['siteurl'] ) {
+					$is_valid = true;
+				}
+			}
+		}
+
+		/**
+		 * Filters whether the sync_error_idc option is valid.
+		 *
+		 * @param bool $is_valid If the sync_error_idc is valid or not.
+		 *
+		 * @since 4.4.0
+		 */
+		$is_valid = (bool) apply_filters( 'jetpack_sync_error_idc_validation', $is_valid );
+
+		if ( ! $is_valid && $sync_error ) {
+			// Since the option exists, and did not validate, delete it.
+			Jetpack_Options::delete_option( 'sync_error_idc' );
+		}
+
+		return $is_valid;
+	}
+
+	/**
+	 * Normalizes a url by doing three things:
+	 *  - Strips protocol
+	 *  - Strips www
+	 *  - Adds a trailing slash
+	 *
+	 * @param string $url URL to parse.
+	 *
+	 * @return WP_Error|string
+	 * @since 4.4.0
+	 */
+	public static function normalize_url_protocol_agnostic( $url ) {
+		$parsed_url = wp_parse_url( trailingslashit( esc_url_raw( $url ) ) );
+		if ( ! $parsed_url || empty( $parsed_url['host'] ) || empty( $parsed_url['path'] ) ) {
+			return new WP_Error(
+				'cannot_parse_url',
+				sprintf(
+				/* translators: %s: URL to parse. */
+					esc_html__( 'Cannot parse URL %s', 'jetpack' ),
+					$url
+				)
+			);
+		}
+
+		// Strip www and protocols.
+		$url = preg_replace( '/^www\./i', '', $parsed_url['host'] . $parsed_url['path'] );
+
+		return $url;
+	}
+
+	/**
+	 * Gets the value that is to be saved in the jetpack_sync_error_idc option.
+	 *
+	 * @param array $response HTTP response.
+	 *
+	 * @return array Array of the local urls, wpcom urls, and error code.
+	 * @since 4.4.0
```

Comment:
these since are incorrect, new functions to the IDC package originally they were in Jetpack class.